include/nbl/ui/CWindowWin32.h

```diff
@@ -16,32 +16,22 @@ class CWindowWin32 final : public IWindowWin32
 	static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 	static E_KEY_CODE getNablaKeyCodeFromNative(uint8_t nativeWindowsKeyCode);
 
-	//TODO
-	//explicit CWindowWin32(CWindowManagerWin32* winManager, core::smart_refctd_ptr<system::ISystem>&& sys, native_handle_t hwnd) : IWindowWin32(std::move(sys))
-	//{
-	//	assert(false);
-	//	RECT rect;
-	//	GetWindowRect(hwnd, &rect);
-	//
-	//	m_width = rect.right - rect.left;
-	//	m_height = rect.bottom - rect.top;
-	//	// TODO m_flags
-	//}
+	CWindowWin32(CWindowManagerWin32* winManager, SCreationParams&& params, native_handle_t hwnd);
 
 	native_handle_t getNativeHandle() const override { return m_native; }
 
-	static core::smart_refctd_ptr<CWindowWin32> create(CWindowManagerWin32* winManager, core::smart_refctd_ptr<system::ISystem>&& sys, int32_t _x, int32_t _y, uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags, const std::string_view& caption)
+	static core::smart_refctd_ptr<CWindowWin32> create(CWindowManagerWin32* winManager, core::smart_refctd_ptr<system::ISystem>&& sys, SCreationParams&& params)
```

Comment:
`CWindow*::create()` should exist any more, only windows manager should be able to create windows